api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java

```diff
@@ -831,8 +831,9 @@ public Mono<ResponseEntity<Void>> deleteProductInInventory(@PathVariable String
     public Flux<ProductResponseDTO> getProductsInInventoryByInventoryIdAndFields(@PathVariable String inventoryId,
                                                                                                  @RequestParam(required = false) String productName,
                                                                                                  @RequestParam(required = false) Double productPrice,
-                                                                                                 @RequestParam(required = false) Integer productQuantity){
-        return inventoryServiceClient.getProductsInInventoryByInventoryIdAndProductsField(inventoryId, productName, productPrice, productQuantity);
+                                                                                                 @RequestParam(required = false) Integer productQuantity,
+                                                                                                 @RequestParam(required = false) Double productSalePrice){
+        return inventoryServiceClient.getProductsInInventoryByInventoryIdAndProductsField(inventoryId, productName, productPrice, productQuantity, productSalePrice);
```

Comment:
Can you explain me why you had to modify this Methodd?

Reply:
Hi brother, I had to modify this method because I needed to add the sale price since I was adding the search by sale price. This allowed searching for the sale price alone and by all the fields combined together. Thanks for asking. Have a good rest of your day Abdur.